src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp

```diff
@@ -660,6 +665,78 @@ static int update(int argc, char *argv[])
     if (shell.empty() && !id.empty())
         return -EINVAL;
 
+    if(!xclbin.empty())
+    {
+
+      std::ifstream in(xclbin);
```

Comment:
xclbin is a binary file, maybe open as such (std::ios::binary)

Reply:
> Instead of adding a new option "xbmgmt flash --xclbin" can we have something like "xbmgmt flash --shell --path"? We already support mcs, dsabin and xsabin file types for this option and I guess your changes in firmware.cpp takes care of supporting xclbin as well.
> According to me, these changes belong in "updateShell(...)", but please let me know if I'm mistaken.
> Can you also add more descriptive comments in the codebase?
> Thanks!



> Instead of adding a new option "xbmgmt flash --xclbin" can we have something like "xbmgmt flash --shell --path"? We already support mcs, dsabin and xsabin file types for this option and I guess your changes in firmware.cpp takes care of supporting xclbin as well.
> According to me, these changes belong in "updateShell(...)", but please let me know if I'm mistaken.
> Can you also add more descriptive comments in the codebase?
> Thanks!



> Instead of adding a new option "xbmgmt flash --xclbin" can we have something like "xbmgmt flash --shell --path"? We already support mcs, dsabin and xsabin file types for this option and I guess your changes in firmware.cpp takes care of supporting xclbin as well.
> According to me, these changes belong in "updateShell(...)", but please let me know if I'm mistaken.
> Can you also add more descriptive comments in the codebase?
> Thanks!

Hi @AShivangi , Thanks for your review. xbmgmt need to verify whether xclbin matches with one of the installed Shells or not. If no installed-Shell is matched, then we shouldnt flash this xclbin. We wanted to make sure that driver load will load the SCHEDULER and CMC firmware from the matched installed Shell.

Reply:
We already have code that does that through "isSameShellOrSC(...)". xsabin and xclbin are similarly structured so it doesn't make sense to have a completely different option for xclbin flashing. I think the flow should go like:

- call updateShell(...) with xclbin file
- if suffix == xclbin, do the checks you are doing
- if valid shell is present, make a firmwareImage object and "upgradeFirmware(...)"

Reply:
I'm still confused with what the entire flow look like. Please give an example flow. Today, xbmgmt either find the file to flash based on VBNV name or flash a file given by path. It seems like we are adding something different here? Feel free to start email discussion on this.

Reply:
Removed extra option "--xclbin" from xbmbmt,
Now user can update flat shell xclbin using "--path" option

Reply:
feedback incorporated.